src/index.ts

```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT. See LICENSE file in the project root for full license information.
 
 
+export * from "./spec";
```

Comment:
```suggestion
export type { CompiledConfig, Config, JsonDocument } from "./spec";
```
Some of these types have no relevance with the other exports of the files. For example, types that support the CLI. 

Likewise, some of the others are internals and I'd like to avoid making them public.